src/main/java/org/apache/mesos/hdfs/executor/NameNodeExecutor.java

```diff
@@ -59,18 +69,27 @@ public void launchTask(final ExecutorDriver driver, final TaskInfo taskInfo) {
           .setTaskId(journalNodeTask.taskInfo.getTaskId())
           .setState(TaskState.TASK_RUNNING)
           .build());
+      healthCheckJN = new TimedHealthCheck(driver, journalNodeTask);
+      timer.scheduleAtFixedRate(healthCheckJN, schedulerConf.getHealthCheckWaitingPeriod(),
+          schedulerConf.getHealthCheckFrequency());
```

Comment:
Instead of building your own health-check setup, how about using TaskInfo.HealthCheck and the TaskStatus.healthy field.
https://github.com/apache/mesos/blob/0.21.1/include/mesos/mesos.proto#L136